tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Builders/UsersControllerBuilder.cs

```diff
@@ -33,15 +27,9 @@ internal UsersControllerBuilder SetUsers(IEnumerable<ApplicationUser> users)
             return this;
         }
 
-        internal UsersControllerBuilder SetOrganisation(Organisation organisation)
-        {
-            _organisation = organisation;
-            return this;
-        }
-
-        internal UsersController Build()
+        internal (UsersController Controller, Mock<IUsersRepository> UserRepository) Build()
         {
-            return new UsersController(_usersRepositoryMock.Object, _organisationMock.Object);
+            return (new UsersController(_usersRepositoryMock.Object), _usersRepositoryMock);
```

Comment:
As briefly discussed this type of pattern I think should be presented to the rest of the team, so just putting down some of my thoughts for discussion.

In my opinion this looks like a bastardization of the builder pattern & a test context approach. It is now building two objects, rather than keeping it simple to just one. The introduction of more dependencies will increase the size of the tuple returning from the build method & could quickly get out of hand.

Even though having the mocked object within the builder makes the test code a bit neater it does lose some flexibility, meaning the test method cannot setup the methods specific to its logic. How could the code test when an exception is raised from the repository?

We all want neater & easy to read tests, so I am not dismissing this approach, just have some reservations, hence why I would like to open a discussion.

Reply:
> In my opinion this looks like a bastardization of the builder pattern & a test context approach. It is now building two objects, rather than keeping it simple to just one. The introduction of more dependencies will increase the size of the tuple returning from the build method & could quickly get out of hand.

**If the tuple gets too big, I see that as an indication that the class itself is doing too much and can probably be broken down**

> Even though having the mocked object within the builder makes the test code a bit neater it does lose some flexibility, meaning the test method cannot setup the methods specific to its logic. How could the code test when an exception is raised from the repository?

**This is not true, you can do built.UserController.Setup().Throws() after you have called Build(), so it's a really clean way of modifying the default built setup to fit test specific needs**